### ISSUE TITLE
add helper to build block explorer link

### ISSUE THREAD

**Issue by LuizAsFight** [FuelLabs/fuels-ts#600]
### Motivation

In many projects, users will need to build a link to block explorer, to show transactions or for any other reason.

we should add a helper method to allow it.

- use a default Block Explorer URL (our latest one -> https://fuellabs.github.io/block-explorer-v2/)
- allow the user to configure other blockExplorerUrl

### Possible implementations

```
export function getBlockExplorerLink({
  path,
  providerUrl,
}: {
  path: string;
  providerUrl: string;
  blockExplorerUrl: string;
}) {
  const blockUrl = blockExplorerUrl || BLOCK_EXPLORER_URL;
  return `${blockUrl}${path}?providerUrl=${encodeURIComponent(
    providerUrl
  )}`;
}
```

**Comment by Dhaiwat10** [comment]
Hey @LuizAsFight, mind if I pick this one up as my first issue? 😄

**Comment by Dhaiwat10** [comment]
Also wanted to ask - which packages out of these do you recommend I add this function to? Or should I just create a new `utils.ts` file in the `fuels` package/folder?

<img width="389" alt="image" src="https://user-images.githubusercontent.com/39617427/203639997-ff502a17-b5c1-4e8a-a190-b8492595dc61.png">

**Comment by LuizAsFight** [comment]
hey @Dhaiwat10 , of course I think it's a good one for you to start.

I would put it in `@fuel-ts/providers/util.ts`

**Comment by arboleya** [comment]
Here's my 2c on the topic.

Although there are many occurrences of this pattern in the repo, I suggest we avoid abstract filenames such as `utils.js` with multiple methods inside and instead favor one `utils` directory with various files inside, containing one function each. 

Smaller files with single responsibilities make it easier to maintain and test. And quicker to find and open (_since filenames mirror their inner function names_).

It also reduces the chances of having multiple files with the same _[abstract]_ name:
 - ![utils](https://user-images.githubusercontent.com/26660/203840632-c5f727d7-44ae-4097-bb21-3059d0bc6ff2.png)

When implemented, it looks something like this:
 - ![tree](https://user-images.githubusercontent.com/26660/203841548-ca443a47-6810-4b2b-8fb5-5a1f2d01a8c6.png)

**Comment by Dhaiwat10** [comment]
@arboleya I agree, and I think having all the utils under a single directory like that will only prove to be more convenient as we add more & more of these utility functions in the future. This directory structure feels like it would be easier to maintain too as a result

**Comment by LuizAsFight** [comment]
I also prefer splitting this `utils` into functional files. But it's a big refactor that will impact a lot of places.

Maybe we can start purging all the `utils` files, but module by module
